spot-webdriver/user/spot-user.js

```diff
@@ -131,6 +156,30 @@ class SpotUser {
             `p2p not established with ${this._internalRemoteControlServiceName}`
         );
     }
+
+    /**
+     * Polls for the XMPP connection to join the control MUC.
+     *
+     * @returns {void}
+     */
+    waitForSignalingConnectionEstablished() {
+        this.driver.waitUntil(
+            () => this.driver.execute(rcsName => {
+                // browser context - you may not access client or console
+                try {
+                    const internalXmppConnection = window.spot[rcsName].xmppConnection;
+
+
+                    return internalXmppConnection.isConnected()
+                        && internalXmppConnection._hasJoinedMuc;
+                } catch (e) {
+                    return false;
+                }
+            }, this._internalRemoteControlServiceName),
+            120000,
```

Comment:
This very long wait currently caused by the fact that it needs to wait for the BOSH session to timeout, right? Is it possible to reference a constant, so that we know which waits to change if we would be moving away from BOSH?

Reply:
The long waits are from absolute frustration from, yes, waiting for the BOSH session timeout. I'll add a constant called THIS_GHOST_PARTICIPANT_WONT_GO_AWAY or something.